### PR TITLE
[Feature Request]: RecordTypeName #214

### LINKED ISSUE #214: [Feature Request]: RecordTypeName
### Is your feature request related to a problem?

Support for record type as described here:
https://dart.dev/language/records

I have some in progress branch with preliminary implementation.

I used it for returning record type from dart methods, but should be generic enough to work everywhere.

### Describe the solution you'd like.

RecordTypeName class which derives from TypeName

### Describe alternatives you've considered.

-

### Other

_No response_

### PR DESCRIPTION
**Pull request by kwasny2**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before
merging your code._

- [ ] I have read the CONTRIBUTING.md
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you
did and what alternatives you considered, etc...

### CODE REVIEW

File: src/main/kotlin/net/theevilreaper/dartpoet/type/RecordTypeName.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        out.emit(")")
+
+        if (isNullable) {
+            out.emit(NULLABLE_CHAR)
+        }
+        return out
+    }
+
+    override fun getRawData(): String = "Record"
```
Comment (theEvilReaper) [inline src/main/kotlin/net/theevilreaper/dartpoet/type/RecordTypeName.kt]: The return value `Record` could raise potential issues. Normally, this method is used to retrieve encapsulated data from a TypeName without brackets, commas, etc.

**Review by theEvilReaper — CHANGES_REQUESTED** [review]
Hey @kwasny2,

it would be great if you could add some tests for `RecordTypeName` to ensure that the generation works and that different edge cases are covered.

What I prefer is having a class like `RecordTypeNameTest` which only contains tests that verify the behaviour of the new type name implementation. After that there should be some tests where record type names are embedded into a `CodeBlock` to check if the processing is valid.

For the isolated test cases, you can use the record examples from the [Dart wiki](https://dart.dev/language/records). You are free to create additional cases that are not covered by the wiki.

**Comment by theEvilReaper** [comment]
See #297 for more details
